from_dict/_caching.py

```diff
@@ -0,0 +1,27 @@
+import functools
+from typing import Optional
+
+from . import _from_dict
+
+_original_function = _from_dict._get_constructor_type_hints
+_cached_function: Optional['functools._lru_cache_wrapper'] = None
+
+def cache_enable(max_size: int = 100):
+    global _cached_function
+    if _cached_function:
+        if _cached_function.cache_info().maxsize == max_size:
+            return
+        _cached_function.cache_clear()
+    _cached_function = functools.lru_cache(max_size)(_original_function)
+    _from_dict._get_constructor_type_hints = _cached_function
+
+
+def cache_disable():
+    global _cached_function
+    if _cached_function:
+        _cached_function.cache_clear()
+        _cached_function = None
+    _from_dict._get_constructor_type_hints = _original_function
+
+def cache_is_enabled() -> bool:
```

Comment:
I think there shouldn't be an option to enable and disable cache. We should be sure it works and then just have it always active. I feel the user shouldn't need to know about it and it's just an internal optimization we do to make it go supafast.
What would be a reason the user would want to disable this?

Reply:
I think I was just overthinking the problem. 😄 

I was worried that you would not want caching at all and I thought if I disabled it by default it would easier to justify the change. It does add a lot of extra code to the package.